### PR TITLE
Clean-up sources ready tracking in kubelet

### PR DESCRIPTION
**Pull request by derekwaynecarr**
moved sources ready tracking behind an interface, made it thread-safe.

### CODE REVIEW

File: pkg/kubelet/config/sources.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import (
+	"sync"
+
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// SourcesReadyFn is function that returns true if the specified sources have been seen.
+type SourcesReadyFn func(sourcesSeen sets.String) bool
+
+// SourcesReady is a simple interface to track what set of configured sources have been seen by the kubelet.
```
Comment (pmorie) [inline pkg/kubelet/config/sources.go]: s/is a simple interface to track what/tracks the/

**Comment by pmorie** [comment]
LGTM aside from the minor nit.

**Comment by yujuhong** [comment]
LGTM. Feel free to apply the label after addressing @pmorie's nits.

**Comment by derekwaynecarr** [comment]
@k8s-bot test this issue https://github.com/kubernetes/kubernetes/issues/17523

**Pull request merged**